src/main/kotlin/org/rust/ide/refactoring/move/common/RsMoveUtil.kt

```diff
@@ -0,0 +1,75 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.ide.refactoring.move.common
+
+import com.intellij.psi.PsiComment
+import com.intellij.psi.PsiElement
+import com.intellij.psi.PsiWhiteSpace
+import org.rust.ide.inspections.import.insertUseItem
+import org.rust.lang.core.psi.RsBlock
+import org.rust.lang.core.psi.RsPath
+import org.rust.lang.core.psi.RsPsiFactory
+import org.rust.lang.core.psi.RsUseItem
+import org.rust.lang.core.psi.ext.*
+
+class RsMoveReferenceInfo(
+    val pathOld: RsPath,
+    val pathNew: RsPath?,
+    // == `pathOld.reference.resolve()`
+    val target: RsQualifiedNamedElement
+) {
+    val isInsideUseDirective: Boolean get() = pathOld.ancestorStrict<RsUseItem>() != null
+}
+
+fun RsPath.isAbsolute(): Boolean {
+    if (text.startsWith("::")) return true
+    if (startsWithSuper()) return false
+
+    val basePathTarget = basePath().reference?.resolve() as? RsMod ?: return false
+    return basePathTarget.isCrateRoot
+}
+
+fun RsPath.startsWithSuper(): Boolean = basePath().referenceName == "super"
+
+// like `text`, but without spaces and comments
+val RsPath.textNormalized: String
+    get() {
+        val copy = copy()
+        for (child in copy.childrenOfType<PsiElement>()) {
+            if (child is PsiWhiteSpace || child is PsiComment) {
+                child.delete()
```

Comment:
Deleting PSI children while iterating looks dangerous.

Also, `childrenOfType` returns only children, but not all descendants. We usually use our methods from `org.rust.lang.core.psi.ext.PsiElement.kt` as they have more precise names. You could use `PsiElement.descendantsOfType()` or even `PsiTreeUtil.findChildrenOfAnyType(copy, PsiWhiteSpace::class.java, PsiComment::class.java)` here. But probably it would be even better to build a string from a sequence of nested `RsPath.path` instead of iterating+deleting PSI

Reply:
Thank you, changed to building a string from parts